documentation/src/docs/controller-service.md

```diff
@@ -341,21 +338,29 @@ Once the Stream Segments are sealed, the Controller needs to store additional in
 
 The following are the Transaction Related metadata records:
 
-   - **Active Transactions**: Each new Transaction is created under the znode. This stores metadata
- corresponding to each Transaction as *Active Transaction Record*. Once a
- Transaction is completed, a new node is created under the global
- _Completed Transaction_ znode and removed from under the Stream specific
- _Active Transaction_ node.
+   - **Active Transactions**: Each new Transaction is created in the a transactions table created for each epoch in the stream. The metadata
+ for each Transaction is stored in an object called *Active Transaction Record*. Once a
+ Transaction is completed, it is moved to a global
+ table for _Completed Transaction_ and its active transaction record is removed.
+
+   - **Completed Transactions**: All completed transactions for all Streams are moved under a separate global table upon completion (via either commit or abort paths). The completion status of Transaction is recorded under this record. To avoid proliferation of stale Transaction records, we provide a cluster level configuration to specify the duration for which a completed Transaction's record should be preserved. Controller periodically garbage collects all Transactions that were completed before the aforesaid configured duration.
 
-   - **Completed Transactions**: All completed transactions for all Streams are moved under a separate znode upon completion (via either commit or abort paths). The completion status of Transaction is recorded under this record. To avoid proliferation of stale Transaction records, we provide a cluster level configuration to specify the duration for which a completed Transaction's record should be preserved. Controller periodically garbage collects all Transactions that were completed before the aforesaid configured duration.
+#### Retention Set
 
+ Controller periodically generates the tail streamcut on the stream and stores them as a chronological time series in the stream metadata called retention set. This retention set is stored in the metadata table. These streamcuts are used to identify truncation points for application of retention policy for the stream. 
+
+#### Writer Marks
+ Writers report their positions and times in the form of writer marks to controller service which are then stored in the metadata table. One of the controller instances then looks at these marks and consolidates them to produce watermarks. 
+
+#### Subscribers
+ As part of 0.9, a new feature has been included in Pravega to keep track of subscriber readergroups and the stream automatically purges the data only after it has been consumed by all subscribers. To achieve this, readergroups report their positions to controller and controller stores their subscriber streamcuts in subscriber metadata for the stream. It then consolidates these streamcuts to compute a lower bound and truncate at such a lowerbound. 
```

Comment:
It may help to explain that this feature is "off" by default and how it could be switched on.